RoomMate.Front/RoomMatePortal/src/app/user-controll-panel/register/register.component.ts

```diff
@@ -1,99 +1,222 @@
+        validators: [Validators.required, Validators.minLength(6)]
+      }),
+    });
+
+    this.citiesFromApi = this.autoComplForm.valueChanges.pipe(
+      startWith(''),
+      map(letters => (letters.length >= 2 ? this.getCities(letters) : []))
+    );
+
+    this.addresCtrl = new FormControl();
+    this.addressFromApi = this.addresCtrl.valueChanges.pipe(
+      startWith(''),
+      map(streetLetters => (streetLetters.length >= 2 && this.cityGetSuccess ? this.getAddress(streetLetters) : []))
+    );
+
+    this.autoComplForm.setValidators(forbiddenNamesValidator(this.pusheditems));
+
+    // on init ends here
   }
 
 
+
+
+
   closeModal(): void {
     this.dialogRef.close();
   }
 
-   // convenience getter for easy access to form fields
-   get f() { return this.form.controls; }
+  // convenience getter for easy access to form fields
+  get f() {
+    return this.form.controls;
+  }
 
   register() {
+    this.selectRole(this.addrSelectSuccess);
     this.disabledButton = true;
     this.registerDto.name = this.form.value.name;
     this.registerDto.surname = this.form.value.surname;
     this.registerDto.email = this.form.value.email;
     this.registerDto.login = this.form.value.login;
     this.registerDto.password = this.form.value.password;
     this.registerDto.roletype = this.registerVariable;
-    debugger;
-    let xd = this.registerDto;
-    this.registerService.register(this.registerDto).subscribe(response => {
-      if (response) {
-        xd = this.registerDto;
-        debugger;
-        this.isRegistered = true;
-        this.registerEvent.emit(this.isRegistered);
-        this.openSnackBar('Register success', 'Ok');
-        this.closeModal();
-      }
-      else {
-        this.isRegistered = false;
-        this.registerEvent.emit(this.isRegistered);
 
-        this.openSnackBar('Register fail', 'Ok');
+    this.registerService.register(this.registerDto).subscribe(
+      response => {
+        if (response) {
+          this.isRegistered = true;
+          this.registerEvent.emit(this.isRegistered);
+          this.openSnackBar('Register success', 'Ok');
+          this.closeModal();
+        } else {
+          this.isRegistered = false;
+          this.registerEvent.emit(this.isRegistered);
+
+          this.openSnackBar('Register failure', 'Ok');
+        }
+        this.disabledButton = false;
+      },
+      error => {
+        this.errorShow = true;
       }
-      this.disabledButton = false;
-    },
-    error => {
-      this.errorShow = true;
-    });
+    );
   }
 
   openSnackBar(message: string, action: string) {
     this._snackBar.open(message, action, {
-      duration: 2000,
+      duration: 2000
     });
   }
 
+  getCities(letters: string): CityDto[] {
+    /// Pierwsze co to wykona się to o:
+    this.pusheditems = new Array<CityDto>();
+    this.registerService.getCityByTwoLetters(letters).subscribe(response => {
+      if (response != null) {
+        response.forEach(element => {
```

Comment:
To mi się nie podoba, sam zobaczę jak to możemy zamienić. Foreach jest tutaj bezsensu jeżeli zwracamy listę z Api

Reply:
ok